### PR TITLE
IRGen: "Simple" partial_apply still needs to re-sign the entry point pointer.

### PR DESCRIPTION
**Pull request by jckarter**
Fixes rdar://problem/63184042

### CODE REVIEW

**Comment by jckarter** [comment]
@swift-ci Please test

File: test/IRGen/simple_partial_apply.sil [hunk 1]
```diff
@@ -13,7 +13,7 @@ struct SingleRefcounted {
 }
 
 // CHECK-LABEL: define {{.*}} @escape_partial_apply_swift_class
-// CHECK:      [[FPTR:%.*]] = insertvalue { i8*, %swift.refcounted* } undef, i8* %0, 0
+// CHECK:      [[FPTR:%.*]] = insertvalue { i8*, %swift.refcounted* } undef, i8* {{.*}}, 0
```
Comment (rjmccall) [inline test/IRGen/simple_partial_apply.sil:16]: Can we test the desired code in at least one of these cases?

Comment (jckarter) [reply test/IRGen/simple_partial_apply.sil:16]: Sure. Does the re-signing code look reasonable?

Comment (rjmccall) [reply test/IRGen/simple_partial_apply.sil:16]: It looks right, yeah.

**Comment by jckarter** [comment]
@swift-ci Please test

**Pull request merged**